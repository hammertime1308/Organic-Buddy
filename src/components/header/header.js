import React from 'react';
import styled from 'styled-components/native';

import { COLOR, FONT } from '../../styles';

const StyledHeader = styled.Text`
  background-color: ${COLOR.GreenHeader};
  padding-top: 10px;
  text-align: center;
  font-size: ${FONT.FontSizeHeader};
  color: ${COLOR.White};
  font-weight: ${FONT.FontWeightBold};
`;

export const Header = ({ children = '' }) => (
  <StyledHeader>{children}</StyledHeader>
);
