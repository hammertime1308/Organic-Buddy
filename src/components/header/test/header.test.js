import React from 'react';
import renderer from 'react-test-renderer';
import { Header } from '../header';

describe('Header', () => {
  it('Should render empty header', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render header with text', () => {
    const tree = renderer.create(<Header>Sample heading</Header>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
