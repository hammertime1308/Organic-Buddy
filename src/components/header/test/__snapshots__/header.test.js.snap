// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should render empty header 1`] = `
<Text
  style={
    Array [
      Object {
        "backgroundColor": "rgba(76, 154, 42, 1)",
        "color": "#ffff",
        "fontSize": 20,
        "fontWeight": "700",
        "paddingTop": 10,
        "textAlign": "center",
      },
    ]
  }
>
  
</Text>
`;

exports[`Header Should render header with text 1`] = `
<Text
  style={
    Array [
      Object {
        "backgroundColor": "rgba(76, 154, 42, 1)",
        "color": "#ffff",
        "fontSize": 20,
        "fontWeight": "700",
        "paddingTop": 10,
        "textAlign": "center",
      },
    ]
  }
>
  Sample heading
</Text>
`;
